src/ui/ui_settings/ui_settings_service_for_request.js

```diff
@@ -0,0 +1,20 @@
+import { uiSettingsServiceFactory } from './ui_settings_service_factory';
+
+const BY_REQUEST_CACHE = new WeakMap();
+
+export function getUiSettingsServiceForRequest(server, request, readInterceptor) {
+  if (BY_REQUEST_CACHE.has(request)) {
```

Comment:
__note:__ I'm not really sure if there will be a big win, but at one of my previous projects we usually tried to reduce two calls (`has` and `get`) to just one `get` in similar cases, so it looked like this. Just wanted to share.
```
let uiSettingsServices = BY_REQUEST_CACHE.get(request);
if (uiSettingsServices) {
    return uiSettingsServices;
}

.....

uiSettingsServices = uiSettingsServiceFactory(....);
.....
```